src/data/abilities/ab-attrs/forewarn-ab-attr.ts

```diff
@@ -3,47 +3,56 @@ import { globalScene } from "#app/global-scene";
 import { getPokemonNameWithAffix } from "#app/messages";
 import { MoveId } from "#enums/move-id";
 import type { Pokemon } from "#field/pokemon";
+import type { PokemonMove } from "#field/pokemon-move";
 import { OneHitKOAttr } from "#moves/one-hit-ko-attr";
 import i18next from "i18next";
 
 export class ForewarnAbAttr extends PostSummonAbAttr {
   public override apply(pokemon: Pokemon, simulated: boolean): boolean {
     let maxPowerSeen = 0;
-    let maxMove = "";
+    let moveName = "";
     let movePower = 0;
+    const movesets: PokemonMove[] = [];
+
     for (const opponent of pokemon.getOpponents()) {
-      for (const move of opponent.moveset) {
-        if (move.getMove().isStatusMove()) {
-          movePower = 1;
-        } else if (move.getMove().hasAttr(OneHitKOAttr)) {
-          movePower = 150;
-        } else if (
-          move.getMove().id === MoveId.COUNTER
-          || move.getMove().id === MoveId.MIRROR_COAT
-          || move.getMove().id === MoveId.METAL_BURST
-        ) {
-          movePower = 120;
-        } else if (move.getMove().power === -1) {
-          movePower = 80;
-        } else {
-          movePower = move.getMove().power;
-        }
+      // TODO: should this consider summon data movesets (such as from Transform)?
+      movesets.push(...opponent.getMoveset(true));
```

Comment:
```suggestion
      movesets.push(...opponent.getMoveset());
```
It should :pikasmile: